scripts/compile-idl-js

```diff
@@ -3,7 +3,47 @@
 # Compiles candid .did files to .idl.js
 
 set -euo pipefail
-find candid/ -type f -name '*.did' |
-  grep -v candid/payloads.did |
-  sed 's/did$//g' |
-  xargs -I{} bash -c '{ echo "/* Do not edit.  Compiled with ./scripts/compile-idl-js from {}did */" ; didc bind -t js "{}did" ; } | sed "s/^export default/export const idlFactory =/g" > "{}"idl.js'
+
+did_files_to_compile() {
+  find candid/ -type f -name '*.did' |
+    grep -v candid/payloads.did
+}
+
+# Normal API access.
+compile_did() {
+  local didfile="$1"
+  local jsfile="$(echo "$didfile" | sed 's/did$/idl.js/g')"
+
+  {
+    echo "/* Do not edit.  Compiled with ./scripts/compile-idl-js from ${didfile} */"
+    didc bind -t js "${didfile}"
+  } | sed "s/^export default/export const idlFactory =/g" >"${jsfile}"
+}
+
+# The certified API makes use of the fact that update calls are always signed, so it calls query calls as updates.
+# Note: This is inefficient if query calls are already certified, which is often but not always the case.
+# Note: It is not clear which API calls have certified responses or how to check that a query response has a certificate.
```

Comment:
On the governance canister I _think_ none of the queries are currently certified. A very high-level check is looking at the signature of the method. If it isn't returning a certificate of some form, then it's definitely not certified.